compiler/rustc_log/src/lib.rs

```diff
@@ -104,7 +115,7 @@ pub fn init_logger(cfg: LoggerConfig) -> Result<(), Error> {
     };
 
     let mut layer = tracing_tree::HierarchicalLayer::default()
-        .with_writer(io::stderr)
+        .with_writer(io::stderr as fn() -> io::Stderr)
```

Comment:
Why is there a fn ptr cast here now?

Reply:
This was a leftover from me trying various interfaces, I removed it now